Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -432,7 +434,56 @@ static MixedContentChecker::ContentType contentTypeFromResourceType(CachedResour
     }
 }
 
-bool CachedResourceLoader::checkInsecureContent(CachedResource::Type type, const URL& url) const
+static MixedContentChecker::IsUpgradable isUpgradableTypeFromResourceType(CachedResource::Type type)
+{
+    switch (type) {
+    // https://www.w3.org/TR/mixed-content/#category-upgradeable
+    // Editor’s Draft, 23 February 2023
+    // 3.1. Upgradeable Content
+    case CachedResource::Type::ImageResource:
+    case CachedResource::Type::MediaResource:
+#if ENABLE(MODEL_ELEMENT)
+    case CachedResource::Type::ModelResource:
+#endif
+        return MixedContentChecker::IsUpgradable::Yes;
```

Comment:
It would read better with something like `return type == Image || Media || ModelResource`.
Maybe we should align with the spec and use options.destination and options.initiator, which seem all available at call site of isUpgradableTypeFromResourceType.

What about the following note from the spec:
`for example, that mixed content images loaded via <img crossorigin ...> will be blocked.`

Reply:
Sure, I'll investigate this. I was following the existing structure of `contentTypeFromResourceType` as a template.

> What about the following note from the spec:
That is handled in MixedContentChecker::shouldUpgradeInsecureContent where it checks `|| mode == FetchOptions::Mode::Cors)`.